core/modules/replica/HttpSqlIndexModule.cc

```diff
@@ -0,0 +1,342 @@
+                numSucceeded++;
+            } else {
+                numFailed[resultSet.extendedStatus]++;
+            }
+        }
+    );
+
+    string error = "job failure code: " + Job::state2string(job->extendedState())
+            + ", success counter: " + to_string(numSucceeded) + ", error counters:";
+    for (auto&& elem: numFailed) {
+        string const extendedStatus = status2string(elem.first);
+        size_t const num = elem.second;
+        error += " " + extendedStatus + ":" + to_string(num);
+    }
+    return error;
+}
+
+
+/**
+ * Translate a result set of a job into a JSON object. 
+ * 
+ * @note ignore errors reported in the input set for now. Only consider
+ * successful result sets.
+ * 
+ * The output JSON object has the following schema:
+ * @code
+ *   <worker>:{
+ *     <table>:{
+ *       <index-key>:{
+ *         "columns":{
+ *           <column>:<number>
+ *         },
+ *         "comment":<string>
+ *       }
+ *     }
+ *   }
+ * @code
+ *
+ * @param jobResultSet The result set to analyze.
+ * @param context The context string for error reporting.
+ * @return The JSON representation of the result.
+ */
+json result2json(SqlJobResult const& jobResultSet, string const& context) {
+    // The cached locations of the fields are computed once during the very first
+    // iteration over result sets. It's assumed all result sets have the same set
+    // of fields.
+    size_t tableNameIdx  = numeric_limits<size_t>::max();
+    size_t keyNameIdx    = numeric_limits<size_t>::max();
+    size_t sequenceIdx   = numeric_limits<size_t>::max();
+    size_t columnNameIdx = numeric_limits<size_t>::max();
+    size_t commentIdx    = numeric_limits<size_t>::max();
+
+    json result;
+    jobResultSet.iterate([&](SqlJobResult::Worker const& worker,
+                             SqlJobResult::Scope const& scope,
+                             SqlResultSet::ResultSet const& resultSet) {
+        // TODO: Ignoring failed or empty results for now. Will decide
+        // what to do about them later.
+        if (resultSet.extendedStatus != ExtendedCompletionStatus::EXT_STATUS_NONE) return;
+        if (not resultSet.hasResult) return;
+
+        // Compute indexes just once and save it for analyzing this and other result sets.
+        if (tableNameIdx == numeric_limits<size_t>::max()) {
+            for (size_t idx = 0, num = resultSet.fields.size(); idx < num; ++idx) {
+                auto&& fieldName = resultSet.fields[idx].name;
+                if      (fieldName == "Table")         { tableNameIdx  = idx; }
+                else if (fieldName == "Key_name")      { keyNameIdx    = idx; }
+                else if (fieldName == "Seq_in_index")  { sequenceIdx   = idx; }
+                else if (fieldName == "Column_name")   { columnNameIdx = idx; }
+                else if (fieldName == "Index_comment") { commentIdx    = idx; }
+            }
```

Comment:
consider defining the strings I guess. Also, maybe you want an else clause, to report an unexpected fieldName?